smartcontractkit/external-adapters-js
# Feature/sc 22208 - Add workflow and post-commit warning when adapter change is not accompanied by changeset

## Description
This change enforces the requirement that all adapter code changes be accompanied by a changeset by adding a failing CI step.

## Changes
* Add Github workflow which fails when a PR has an adapter change but does not include a changeset
* Add post-commit warning when a commit contains an adapter change but does not include a changeset



## Steps to Test
1. Pull down `feature/sc-22208` and check out a test branch
2. Add a commit to the test branch containing a change to a file in an adapter directory (packages/sources/*/src/* OR packages/composites/*/src/* OR packages/targets/*/src/*)
3. Observe that a post-commit warning appears "⚠️ NOTICE: Adapter changes must also have a changeset. Generate a changeset with 'yarn changeset' and include it in your commit."
4. Push the test branch to Github
5. Observe that the CI fails on step "Adapter changes accompanied by a changeset"

Example PR with failure: https://github.com/smartcontractkit/external-adapters-js/pull/1526

## Quality Assurance
